### DISCUSSION TITLE
Meeting Minutes for September 5, 2024

### DISCUSSION THREAD

**Discussion by dezidizon** [CVWRSM/cvwrsm#113] [Meeting Minutes]
**Agenda:**

- Discussion regarding dropins for WRIMS GUI. Tracked in #112 
- Discussion of Tom's use of Tattletale for OSGI analysis.
- Handling 3rd party dependencies in a 3rd party OSGI bundle for WRIMS.
- Reporting on team to help with the CBC solver.

**Open Action Items:**
- @dwr-ycheng to follow-up on whether RMA should move ahead with preparing the installer for the HEC-DSS Excel Addin.
- @rma-psmorris to organize team to help with the CBC Solver.
- RMA to follow-up with DWR to discuss work for the CBC Solver.
- Prepare level of effort, budget, and schedule to perform work for the following projects:
    - IWFM and IDC Automated Build System and Source Control
    - WRIMS 3 Development

**Discussions:** 
-  @inazrul invited @dwr-zroy, @ekblad, @HamedZamanisabzi to join the bi-weekly status meeting as they are joining the CalSim development team. 
- @hxiedwr discussed that inconsistent behavior would be observed when updating dropins for WRIMS GUI. He hoped the automated build process helps resolve this issue. This is tracked in #112.
    -  To resolve this issue, the cache must be cleared. A cache refresh can be triggered by increasing the version number or a build number.
    - @hxiedwr will prepare a manual item without any function that can be used to reproduce the issue and to verify that it is resolved in the automatic build.
- @dwr-psandhu discussed that the code for the CBC solver needs to be migrated into GitHub and that this should be part of WRIMS. @inazrul wants RMA to be engaged with the process.
    - @HamedZamanisabzi and @ekblad are working on the internal code and publishing on GitHub. 
    - C++ development is not currently occurring. 
    - @rma-psmorris is putting together a team that includes @zack-rma and Ryan Ripken to help with the CBC solver work.
- @rma-tom demonstrated his use of the [Tattletale](https://tattletale.jboss.org/) tool to perform OSGI analysis for WRIMS.
    -  Compared to jdeps, Tattletale makes it possible to look at the contents of a folder and to look at all the jars in a folder. 
    - Tattletale was used to analyze the OSGI plugins and helped create the readme file and Plant UML diagram for WRIMS.
    - This was also used to verify that the wrims plugin jar produced through the automatic build process matches the wrims plugin jar produced by @hxiedwr.
- The 3rd party dependencies for WRIMS will be produced in a 3rd party OSGI bundle.

**Comment by dwr-psandhu** [comment]
I have been working closely with @HamedZamanisabzi  and @ekblad  to investigate jCbc and CBC solver and we have 
 * compiled Babek’s version (7yrs ago) with latest CBC solver (2.10.12) and reproduced the results from the examples
 * Updated the changes made to jCbc internally that were never committed to GitHub and now compiling against latest Cbc ( couple of dead code and unused functions and function signature changes)
 
 @ekblad and @HamedZamanisabzi can give updates

**Reply by dwr-hzamanisabzi** [reply]
Thanks, @dwr-psandhu and Nicky. I just want to add one more update: after compiling the latest code, we are now troubleshooting any potential errors that occur when WRIMS uses the updated compiled jCbc.dll and COINOR.jar files.

**Reply by adamkorynta** [reply]
Couple questions: 

Which repository is jCbc under? 
Is CentralValleyModeling maintaining a fork? 
Also does it compile with newer JDK versions?

**Reply by dwr-hzamanisabzi** [reply]
We are maintaining a fork and will commit all changes after completing the troubleshooting. We are using JDK 1.8, which is compatible with our target applications within WRIMS.

**Reply by dwr-psandhu** [reply]
My original fork was https://github.com/dwr-psandhu/jCbc which works with the original jCbc and using latest Cbc. @HamedZamanisabzi fork would be where jCbc is being updated and is based off this fork.
There is a cmake build that needs cmake 3.30+, swig 4.x and vs2022 along with intel oneapi for compilation.

**Reply by ekblad** [reply]
@HamedZamanisabzi and I reproduced @dwr-psandhu's build so far. AFAIK there is no public fork yet but Nicky's repo is self-contained outside of jCbc files in [CMakelists.txt](https://github.com/dwr-psandhu/jCbc/blob/master/CMakelists.txt). the jCbc.i and jCbc.cpp are used to customize some Cbc functions [here](https://github.com/dwr-psandhu/jCbc/blob/92b8d5a820471996d853f3c726fa74d98ee3e335/CMakelists.txt#L31). can the Cbc 2.10.12 libs be used for these functions in place of jCbc?
